tex2wiki/src/tex2wiki.py

```diff
@@ -0,0 +1,413 @@
+
+
+def generate_nav_bar(info):
+    links = list()
+    for link, text in info:
+        link = link.replace("''", "")
+        text = text.replace("''", "")
+        links.append(generate_link(link, text))
+
+    nav_section = generate_html("div", "<< " + links[0], options={"id": "alignleft"}, spacing=1) + "\n" + \
+                  generate_html("div", links[1], options={"id": "aligncenter"}, spacing=1) + "\n" + \
+                  generate_html("div", links[2] + " >>", options={"id": "alignright"}, spacing=1)
+
+    header = generate_html("div", nav_section, options={"id": "drmf_head"})
+    footer = generate_html("div", nav_section, options={"id": "drmf_foot"})
+
+    return header, footer
+
+
+def get_macro_name(macro):
+    # (str) -> str
+    """Obtains the macro name."""
+    macro_name = ""
+    for ch in macro:
+        if ch.isalpha() or ch == "\\":
+            macro_name += ch
+        elif ch in ["@", "{", "["]:
+            break
+
+    return macro_name
+
+
+def find_all(pattern, string):
+    # (str, str) -> generator
+    """Finds all instances of pattern in string."""
+
+    i = string.find(pattern)
+    while i != -1:
+        yield i
+        i = string.find(pattern, i + 1)
+
+
+def get_symbols(text, glossary):
+    # (str, dict) -> str
+    """Generates span text based on symbols present in text."""
+    symbols = set()
+
+    for keyword in glossary:
+        for index in find_all(keyword, text):
+            # if the macro is present in the text
+            if index != -1:
+                index += len(keyword)  # now index of next character
+                if index >= len(text) or not text[index].isalpha():
+                    symbols.add(keyword)
+
+    span_text = ""
+    for symbol in sorted(symbols, key=str.lower):
+        links = list()
+        for cell in glossary[symbol]:
+            if "http://" in cell or "https://" in cell:
+                links.append(cell)
+
+        id_link = links[0]
+        links = ["[" + link + " " + link + "]" for link in links]
+
+        meaning = list(glossary[symbol][1])
+
+        count = 0
+        for i, ch in enumerate(meaning):
+            if ch == "$" and count % 2 == 0:
+                meaning[i] = "<math>{\\displaystyle "
+                count += 1
+            elif ch == "$":
+                meaning[i] = "}</math>"
+                count += 1
+
+        appearance = glossary[symbol][4].strip("$")
+
+        span_text += "<span class=\"plainlinks\">[" + id_link + " <math>{\\displaystyle " + appearance + \
+                     "}</math>]</span> : " + ''.join(meaning) + " : " + " ".join(links) + "<br />\n"
+
+    return span_text[:-7]  # slice off the extra br and endline
+
+
+def create_general_pages(data, title):
+    ret = ""
+
+    # get list of section names
+    section_names = [d[0] for d in data]
+    section_names = ["Orthogonal Polynomials"] + section_names + ["Orthogonal Polynomials"]
```

Comment:
Actually, I have to change that string - shouldn't be Orthogonal Polynomials in this case. And, it's because the pages go in a loop, and to create the nav bar, there should be a link to the page ahead and the page behind. So, I prepend and append so that when I'm generating links for the very last section in section_names, it doesn't crash, instead looping around. If I instead do a modulus on the indices when accessing data from section_data, it makes the lines too long. This is more legible imo